src/client/datascience/jupyter/kernels/kernel.ts

```diff
@@ -748,3 +786,10 @@ export function getPlainTextOrStreamOutput(outputs: nbformat.IOutput[]) {
     }
     return;
 }
+
+// Options for error reporting from kernel silent execution
+export type SilentExecutionErrorOptions = {
+    traceErrors?: boolean;
+    traceErrorsMessage?: string;
+    logTelemetryErrors?: boolean;
```

Comment:
Should we send the name of the telemetry event here?
This way we have finer grained telemetry, e.g. we can tell from the calling code whether to log telemetry for startup code, user startup code, for variable fetching, etc